contracts/MLMSystem.sol

```diff
@@ -0,0 +1,159 @@
+    uint8 constant feePercents = 5;
+
+    struct Referral {
+        address addressReferral;
+        uint8 level;
+    }
+
+    //Checks if the user is logged into the system
+    modifier notSignedUp() {
+        require(!isSignedUp[msg.sender]);
+        _;
+    }
+
+    // Returns the balance for a specific address
+    function getBalance(address _member) public view returns (uint256 balance) {
+        return userBalance[_member];
+    }
+
+    //Returns the level of the user, depending on his balance
+    function getLevel(address _member) public view returns (uint8) {
+        uint currentBalance = getBalance(_member);
+        if (currentBalance >= 5e15 && currentBalance < 1e16) {
+            return 1;
+        } else if (currentBalance >= 1e16 && currentBalance < 2e16) {
+            return 2;
+        } else if (currentBalance >= 2e16 && currentBalance < 5e16) {
+            return 3;
+        } else if (currentBalance >= 5e16 && currentBalance < 1e17) {
+            return 4;
+        } else if (currentBalance >= 1e17 && currentBalance < 2e17) {
+            return 5;
+        } else if (currentBalance >= 2e17 && currentBalance < 5e17) {
+            return 6;
+        } else if (currentBalance >= 5e17 && currentBalance < 1e18) {
+            return 7;
+        } else if (currentBalance >= 1e18 && currentBalance < 2e18) {
+            return 8;
+        } else if (currentBalance >= 2e18 && currentBalance < 5e18) {
+            return 9;
+        } else if (currentBalance >= 5e18) {
+            return 10;
+        }
+    }
+
+    // Returns an array of structures containing the level and address of the referrals
+    function getReferalsInfo() external view returns (Referral[] memory) {
+        require(directPartners[msg.sender].length != 0, "No referrals ");
+        uint count = directPartners[msg.sender].length;
+        Referral[] memory referralInfo = new Referral[](count);
+        for (uint i = 0; i < directPartners[msg.sender].length; i++) {
+            referralInfo[i] = Referral({
+                addressReferral: directPartners[msg.sender][i],
+                level: getLevel(directPartners[msg.sender][i])
+            });
+        }
+        return (referralInfo);
+    }
+
+    // Registration function without referral link
+    function signUp() public notSignedUp {
+        referralToReferrer[msg.sender] = address(0);
+        isSignedUp[msg.sender] = true;
+    }
+
+    // Referral link registration function
+    function signUp(address _referrer) public notSignedUp {
+        require(isReferrerExist(_referrer), "referrer address doesn't exist");
+        directPartners[_referrer].push(msg.sender);
+        referralToReferrer[msg.sender] = _referrer;
+        isSignedUp[msg.sender] = true;
+    }
+
+    // Function to check the existence of the specified referrer
+    function isReferrerExist(address _referrer) private view returns (bool) {
+        return isSignedUp[_referrer];
+    }
```

Comment:
Order of Functions[](https://docs.soliditylang.org/en/develop/style-guide.html#order-of-functions)